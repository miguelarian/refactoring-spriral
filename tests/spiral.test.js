const spiral = require('../src/spiral')

describe('spiral of size < 3', () => {
    it('should throw an error for 0', () => {
        expect(() => spiral(0)).toThrowError('Size must be odd number >= 3')
    })

    it('should throw an error for 1', () => {
        expect(() => spiral(1)).toThrowError('Size must be odd number >= 3')
    })

    it('should throw an error for 2', () => {
        expect(() => spiral(2)).toThrowError('Size must be odd number >= 3')
    })
})

describe('spiral of size to be odd number > 3', () => {
    it('should throw an error for 4', () => {
        expect(() => spiral(4)).toThrowError('Size must be odd number >= 3')
    })
})

describe('spiral of size', () => {
    ;[3, 5, 7, 9].forEach((size) => {
        it(`should print spiral of size ${size}`, () => {
            let outputData = ''
            const storeLog = (inputs) => (outputData += inputs)
            console['log'] = jest.fn(storeLog)
            spiral(size)
            expect(outputData).toMatchSnapshot()
        })
    })
})
