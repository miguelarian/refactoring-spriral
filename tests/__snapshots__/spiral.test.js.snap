// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spiral of size should print spiral of size 3 1`] = `" 7 8 9 6 1 2 5 4 3"`;

exports[`spiral of size should print spiral of size 5 1`] = `" 21 22 23 24 25 20  7  8  9 10 19  6  1  2 11 18  5  4  3 12 17 16 15 14 13"`;

exports[`spiral of size should print spiral of size 7 1`] = `" 43 44 45 46 47 48 49 42 21 22 23 24 25 26 41 20  7  8  9 10 27 40 19  6  1  2 11 28 39 18  5  4  3 12 29 38 17 16 15 14 13 30 37 36 35 34 33 32 31"`;

exports[`spiral of size should print spiral of size 9 1`] = `" 73 74 75 76 77 78 79 80 81 72 43 44 45 46 47 48 49 50 71 42 21 22 23 24 25 26 51 70 41 20  7  8  9 10 27 52 69 40 19  6  1  2 11 28 53 68 39 18  5  4  3 12 29 54 67 38 17 16 15 14 13 30 55 66 37 36 35 34 33 32 31 56 65 64 63 62 61 60 59 58 57"`;
